contracts/token/ERC1155/extensions/ERC1155Supply.sol

```diff
@@ -31,18 +31,15 @@ abstract contract ERC1155Supply is ERC1155 {
     }
 
     /**
-     * @dev See {ERC1155-_beforeTokenTransfer}.
+     * @dev See {ERC1155-_update}.
      */
-    function _beforeTokenTransfer(
-        address operator,
+    function _update(
         address from,
         address to,
         uint256[] memory ids,
         uint256[] memory amounts,
         bytes memory data
     ) internal virtual override {
-        super._beforeTokenTransfer(operator, from, to, ids, amounts, data);
-
         if (from == address(0)) {
             for (uint256 i = 0; i < ids.length; ++i) {
                 _totalSupply[ids[i]] += amounts[i];
```

Comment:
Here and below we can use `_unsafeAccess` as well.

Reply:
I meant unsafe access for the arrays. I agree that the arithmetic here should not be unchecked.